### ISSUE TITLE
Segfault with --synchronize on Onedrive for Busines Library

### ISSUE THREAD

**Issue by elhennig** [abraunegg/onedrive#998]
**Note:** Before submitting a bug report, please ensure you are running the latest 'onedrive' client as built from 'master' and by using the latest available DMD compiler. Refer to the readme on building the client for your system.

### Bug Report Details ###
**Describe the bug**
The initial sync works fine, but any additional sync try will segfault.

**Application and Operating System Details:**
*   Linux l4pfnhr2 5.3.18-lp152.26-default #1 SMP Mon Jun 29 14:58:38 UTC 2020 (2a0430f) x86_64 x86_64 x86_64 GNU/Linux
Opensuse Leap 15.2
*   Not using gui
*   Onedrive for Business
*   Did you build from source or install from a package? Yes, git checked out today
*   If you installed from source, what is your DMD or LDC compiler version: `dmd --version` or `ldmd2 --version`
dmd --version
DMD64 D Compiler v2.085.1
Copyright (C) 1999-2019 by The D Language Foundation, All Rights Reserved written by Walter Bright

*   Application configuration: Output of `onedrive --display-config`
*   Curl Version: Output of `curl --version`
onedrive --confdir .config/onedrive_business_docs/ --display-config
Configuration file successfully loaded
```
onedrive version                       = v2.4.3-14-g40ee78e
Config path                            = .config/onedrive_business_docs/
Config file found in config path       = true
Config option 'check_nosync'           = false
Config option 'sync_dir'               = /data/e8864/Dokumente/Sharepoint/Dokumente
Config option 'skip_dir'               = 
Config option 'skip_file'              = ~*
Config option 'skip_dotfiles'          = false
Config option 'skip_symlinks'          = false
Config option 'monitor_interval'       = 45
Config option 'min_notify_changes'     = 5
Config option 'log_dir'                = /var/log/onedrive/
Config option 'classify_as_big_delete' = 1000
Config option 'drive_id'               = b!0ID4grHmNkOikNa_F4vY1R0mqni7cf9MrVfPi1vpEqEm3iNDU7dHQYb3moYRRYxk
Config option 'sync_root_files'        = false
Selective sync 'sync_list' configured  = false
Business Shared Folders configured     = false
```
**Note:** Please generate a full debug log as per [https://github.com/abraunegg/onedrive/wiki/Generate-debug-log-for-support](https://github.com/abraunegg/onedrive/wiki/Generate-debug-log-for-support) and email to support@mynas.com.au

**To Reproduce**
Steps to reproduce the behavior if not causing an application crash:
1.  Initial Sync Ondrive for Business Folder
2.  Run synchronize again

NOTE: The problem does not appear with another library below the master library where the problem occurs.

**Complete Verbose Log Output**
A clear and full log of the problem when running the application in the following manner (ie, not in monitor mode):
```bash
onedrive --synchronize --verbose <any of your other needed options>
```
> cat debug_output.log 
```
[DEBUG] homePath: HOME environment variable set
[DEBUG] homePath: /home/e8864
[DEBUG] configDirName: CLI override to set configDirName to: .config/onedrive_business_docs
Using 'user' Config Dir: .config/onedrive_business_docs
Using 'system' Config Dir: 
[DEBUG] refreshTokenFilePath = .config/onedrive_business_docs/refresh_token
[DEBUG] deltaLinkFilePath = .config/onedrive_business_docs/delta_link
[DEBUG] databaseFilePath = .config/onedrive_business_docs/items.sqlite3
[DEBUG] databaseFilePathDryRun = .config/onedrive_business_docs/items-dryrun.sqlite3
[DEBUG] uploadStateFilePath = .config/onedrive_business_docs/resume_upload
[DEBUG] userConfigFilePath = .config/onedrive_business_docs/config
[DEBUG] syncListFilePath = .config/onedrive_business_docs/sync_list
[DEBUG] systemConfigFilePath = /config
[DEBUG] businessSharedFolderFilePath = .config/onedrive_business_docs/business_shared_folders
Configuration file successfully loaded
[DEBUG] sync_dir: Getting syncDir from config value sync_dir
[DEBUG] syncDir: /data/e8864/Dokumente/Sharepoint/Dokumente
[DEBUG] Testing network to ensure network connectivity to Microsoft OneDrive Service
Initializing the OneDrive API ...
Configuring Global Azure AD Endpoints
[DEBUG] Downgrading all HTTP operations to HTTP/1.1 by default
[DEBUG] clientId    = d50ca740-c83f-4d1b-b616-12c519384f0c
[DEBUG] companyName = abraunegg
[DEBUG] appTitle    = OneDrive Client for Linux
Opening the item database ...
[DEBUG] Using database file: .config/onedrive_business_docs/items.sqlite3
All operations will be performed in: /data/e8864/Dokumente/Sharepoint/Dokumente
[DEBUG] Configuring skip_dir ...
[DEBUG] skip_dir: 
[DEBUG] Configuring skip_dir_strict_match ...
[DEBUG] skip_dir_strict_match: false
[DEBUG] Configuring skip_dotfiles ...
[DEBUG] skip_dotfiles: false
[DEBUG] Configuring skip_file ...
[DEBUG] skip_file: ~*
```

Followed by the error output: "Speicherzugriffsfehler (Speicherabzug geschrieben)" (segmentation fault)

**Comment by abraunegg** [comment]
@elhennig 
This is a duplicate for #995 

To resolve:
```
git clone https://github.com/abraunegg/onedrive.git
cd onedrive
git fetch origin pull/997/head:pr997
git checkout pr997
./configure; make clean; make;

```

**Comment by abraunegg** [comment]
@elhennig 
If you feel that this is not a duplicate, please provide a full verbose debug log via email for analysis.

**Comment by elhennig** [comment]
I thought it might be a duplicate, but it behaves slightly differently. I have two libraries to sync and one works while the other (the default one) does not.
I noticed that the items.sqlite3 did not contain any lines in the item table in the not working config. I tried to restart from scratch by removing the  items.sqlite3 and refresh_token from the confdir. Then I started from scratch. 

After some retries it seems to work, now without me knowing what I did to solve it. Very strange! Thanks anyways!
